### ISSUE TITLE
Connection SNI settings?

### ISSUE THREAD

**Issue by codepope** [IBM/etcd-java#4]
As far as I can tell, the client doesn't set the SNI/servername header on its TLS/SSL connections. Where Compose etcd users TLS/SSL it requires that to be set so it can verify with the right certificate. 

Currently working around with `.withTrustManager(InsecureTrustManagerFactory.INSTANCE)` in the connection.

**Comment by njhill** [comment]
Hi @codepope, how were you trying to configure the client and what error were you hitting? To work with compose TLS, you need to either use
```java
.withTrustManager(new ComposeTrustManagerFactory(any_name, compose_deployment_name, cert))
```
, where `cert` can be a specific self-signed pem-format cert or `null` to use JRE default certs, _or_
```
EtcdClusterConfig.fromJsonFile(filePath).getClient();
```
where filePath is the path to a file in [this json format](https://github.com/IBM/etcd-java/blob/master/etcd-json-schema.md) which has its `compose` attribute set to the deployment name

**Comment by codepope** [comment]
You know, I didn't see a mention of a ComposeTrustManagerFactory anywhere, but I will give it a try in the AM and see how it goes. I'll close for now and reopen if no joy.

**Comment by njhill** [comment]
I don't blame you! I will improve the doc for that - only the second approach of the above two is doc'd right now. I'll also add some friendlier constructors to `ComposeTrustManagerFactory`, e.g. one which takes _only_ the deployment name.

**Comment by codepope** [comment]
Ok, yes, I can get it working but ... we need to talk. The fact you are getting the self-signed cert indicates SNI isn't set, so it's doing a verification on the DN/CN name (which is reasonable with no cert) but it can't lean on the Let's Encrypt certs of newer etcds on Compose.

**Comment by njhill** [comment]
The inconsistency in using `ComposeTrustManagerFactory` with `.forEndpoint()` and `.forEndpoints()` has been addressed in the latest [0.0.3 release](https://github.com/IBM/etcd-java/releases/tag/v0.0.3).
